storehaus-leveldb/src/main/scala/com.twitter.storehaus.leveldb/LevelDBStore.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2015 Twitter inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.twitter.storehaus.leveldb
+
+import java.io.File
+import java.util.concurrent.Executors
+
+import com.twitter.storehaus.Store
+import com.twitter.util.{FuturePool, Time, Future, Duration}
+import org.iq80.leveldb._
+import org.fusesource.leveldbjni.JniDBFactory._
+
+/**
+ * Store dealing with a LevelDB database.
+ * @author Ben Fradet
+ * @since 10/03/15
+ */
+class LevelDBStore(val dir: File, val options: Options, val numThreads: Int)
+    extends Store[Array[Byte], Array[Byte]] {
```

Comment:
Actually, I tend to prefer wrapping as close to the underlying store as possible and using combinators (I know @ianoc has some information that can cause problems, but otherwise we have a ton of duplication, which I also think is a problem). What do you think of this comment, @rubanm ?